File: src/sdk/javascript/examples/streaming.ts
/**
 * 2api.ai JavaScript SDK - Streaming Example
 *
 * Demonstrates streaming responses for real-time output.
 */

import { TwoAPI } from 'twoapi';

async function main() {
  const client = new TwoAPI();

  // ============================================================
  // Simple Streaming
  // ============================================================
  console.log('=== Simple Streaming ===\n');

  process.stdout.write('Response: ');
  for await (const chunk of client.chatStream('Tell me a short story about a robot')) {
    process.stdout.write(chunk.content);
    if (chunk.done) {
      console.log('\n[Stream complete]');
    }
  }
  console.log();

  // ============================================================
  // OpenAI-Compatible Streaming
  // ============================================================
  console.log('=== OpenAI-Compatible Streaming ===\n');

  const stream = await client.chat.completions.create({
    model: 'openai/gpt-4o',
    messages: [{ role: 'user', content: 'Count from 1 to 10, one number per line' }],
    stream: true,
  });

  process.stdout.write('Counting: ');
  for await (const chunk of stream) {
    const content = chunk.choices[0]?.delta?.content || '';
    process.stdout.write(content);

    if (chunk.choices[0]?.finishReason) {
      console.log(`\n[Finished: ${chunk.choices[0].finishReason}]`);
    }
  }
  console.log();

  // ============================================================
  // Streaming with Options
  // ============================================================
  console.log('=== Streaming with Temperature ===\n');

  process.stdout.write('Creative response: ');
  for await (const chunk of client.chatStream('Give me a creative name for a coffee shop', {
    model: 'anthropic/claude-3-5-sonnet',
    temperature: 0.9,
    maxTokens: 50,
  })) {
    process.stdout.write(chunk.content);
  }
  console.log('\n');

  // ============================================================
  // Collecting Stream Results
  // ============================================================
  console.log('=== Collecting Stream Results ===\n');

  let fullContent = '';
  let tokenCount = 0;

  for await (const chunk of client.chatStream('What are the three primary colors?')) {
    fullContent += chunk.content;
    tokenCount++;
  }

  console.log('Full response:', fullContent);
  console.log('Chunks received:', tokenCount);
}

main().catch(console.error);

File: src/sdk/python/examples/streaming.py
"""
2api.ai Python SDK - Streaming Example

Demonstrates streaming responses for real-time output.
"""

import sys
from twoapi import TwoAPI

def main():
    client = TwoAPI()

    # ============================================================
    # Simple Streaming
    # ============================================================
    print("=== Simple Streaming ===\n")

    sys.stdout.write("Response: ")
    for chunk in client.chat_stream("Tell me a short story about a robot"):
        sys.stdout.write(chunk.content)
        sys.stdout.flush()
    print("\n[Stream complete]\n")

    # ============================================================
    # OpenAI-Compatible Streaming
    # ============================================================
    print("=== OpenAI-Compatible Streaming ===\n")

    stream = client.chat.completions.create(
        model="openai/gpt-4o",
        messages=[{"role": "user", "content": "Count from 1 to 10, one number per line"}],
        stream=True,
    )

    sys.stdout.write("Counting: ")
    for chunk in stream:
        content = chunk.choices[0].delta.content or ""
        sys.stdout.write(content)
        sys.stdout.flush()

        if chunk.choices[0].finish_reason:
            print(f"\n[Finished: {chunk.choices[0].finish_reason}]")
    print()

    # ============================================================
    # Streaming with Options
    # ============================================================
    print("=== Streaming with Temperature ===\n")

    sys.stdout.write("Creative response: ")
    for chunk in client.chat_stream(
        "Give me a creative name for a coffee shop",
        model="anthropic/claude-3-5-sonnet",
        temperature=0.9,
        max_tokens=50,
    ):
        sys.stdout.write(chunk.content)
        sys.stdout.flush()
    print("\n")

    # ============================================================
    # Collecting Stream Results
    # ============================================================
    print("=== Collecting Stream Results ===\n")

    full_content = ""
    chunk_count = 0

    for chunk in client.chat_stream("What are the three primary colors?"):
        full_content += chunk.content
        chunk_count += 1

    print(f"Full response: {full_content}")
    print(f"Chunks received: {chunk_count}")


if __name__ == "__main__":
    main()
